pkg/provisioner/fixture/fixture.go

```diff
@@ -28,22 +28,25 @@ type fixtureProvisioner struct {
 	publisher provisioner.EventPublisher
 	// state to manage the two-step adopt process
 	adopted bool
+	// status of the provisioner
+	ready bool
 }
 
 // New returns a new Ironic Provisioner
 func New(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publisher provisioner.EventPublisher) (provisioner.Provisioner, error) {
```

Comment:
This function is public and you can add whatever arguments you like, so maybe rather than making the type public you could allow the caller to pass something here (e.g. number of times to return false before returning true from IsReady(), or a user-supplied function to call from IsReady() to get the result).

Reply:
Unfortunately the `New` function must match the `provisioner.Factory` type so I don't think it's recommended to modify its signature. Alternatively I could add another factory method (ie `NewExt` or some other better name), to be used only from the test, with the required additional params, and keep the type private.

Reply:
I've added a new factory to allow injecting test parameters. With this approach, the fixture remains private and no need to move the file from its original location